### PR TITLE
Implement tools for activating a key and RSA signing and more

### PR DESCRIPTION
**Pull request by stefanberger**
Implement tools for activating a key and RSA signing.

Unfortunately the signature the pykmip server is sending back doesn't seem to be verifyable with openssl. I would blame it on the pykmip server, which is the only server I have to test with. The following should work, but doesn't:

```
#> ./bin/sign -cacert /etc/pykmip/certs/server_ca_cert.pem -cert /etc/pykmip/certs/client_cert.pem \
  -key /etc/pykmip/certs/client_private_key.pem -keyid 8 -skipVerify \
  -server <server IP>:5696 -infile text -outfile signature.out -sha256
#> openssl dgst -verify pubkey.pem -signature signature.out -sha256 text
Verification Failure
```

Note: An RSA key for signing can only be created by  modifying the tools that come with the server and setting the key usage flag for signing.

### CODE REVIEW

File: kmip/objects.go [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+// Object models a KMIP object identified by its unique id and located on a server accessible
+// with the given TLSConfig
+type Object struct {
+	UniqueIdentifier *String
+	TLSConfig        *tls.Config
+	Server           *string
+}
+
+// NewObject creates a new Object
+func NewObject(id, server *string, tlsConfig *tls.Config) (*Object, error) {
```
Comment (lumjjb) [inline kmip/objects.go]: are server and tlsConfig optional? - or maybe add a check for non-nil ?

File: kmip/consts.go [hunk 2]
```diff
@@ -206,4 +221,34 @@ const (
 	KRT_PVKIBM   = 0x13
 	KRT_PVKPVV   = 0x14
 	KRT_PVKOTH   = 0x15
+
+	/* Key Compression Types */
+	KCT_None                               = 0x00000000
```
[the remark this answers is not in the document]
Comment (stefanberger) [reply kmip/consts.go]: Basically there should be a Pad_None = 0x0, meaning that the function should not provide any field in the request related to padding, and there's the existing defined Pad_NONE = 0x1, which actually has a meaning. Basically I am missing several XYZ_None = 0x0...

File: kmip/objects.go (hunk 1 above)
Comment (stefanberger) [reply kmip/objects.go]: Server is mandatory and cannot be nil, tlsConfig may be nil, at least theoretically. Maybe there's a server that allows plain TCP connections, but I doubt it.

File: kmip/consts.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (lumjjb) [reply kmip/consts.go]: Ah ok, that makes sense.

**Pull request merged**